kernel/ptrace.c

```diff
@@ -0,0 +1,243 @@
+#include "ptrace.h"
+#include "kernel/calls.h"
+#include "kernel/errno.h"
+#include "kernel/signal.h"
+#include "task.h"
+#include <string.h>
+
+// Returns stopped child with the given pid, locked with stop_lock
+static struct task *find_child(struct task *parent, pid_t_ pid) {
+    struct task *child = NULL;
+    list_for_each_entry(&parent->children, child, siblings) {
+        if (child->pid == pid) {
+            lock(&child->ptrace.stop_lock);
+            if (child->ptrace.stopped) {
+                break;
+            } else {
+                unlock(&child->ptrace.stop_lock);
+            }
+        }
+    }
+    return child;
+}
+
+dword_t sys_ptrace(dword_t request, dword_t pid, addr_t addr, dword_t data) {
+    switch (request) {
+    case PTRACE_TRACEME_:
```

Comment:
Please indent the case label and case body by one more level, and add a blank line after each case, to be consistent with the rest of the project